modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/item/selector/ContentDashboardItemTypeItemSelectorView.java

```diff
@@ -88,12 +84,42 @@ public void renderHTML(
 			PortletURL portletURL, String itemSelectedEventName, boolean search)
 		throws IOException, ServletException {
 
-		_itemSelectorViewDescriptorRenderer.renderHTML(
-			servletRequest, servletResponse,
-			contentDashboardItemTypeItemSelectorCriterion, portletURL,
-			itemSelectedEventName, search,
-			new ContentDashboardItemTypeItemSelectorViewDescriptor(
-				(HttpServletRequest)servletRequest, portletURL));
+		PortletRequest portletRequest =
+			(PortletRequest)servletRequest.getAttribute(
+				JavaConstants.JAVAX_PORTLET_REQUEST);
+
+		PortletResponse portletResponse =
+			(PortletResponse)servletRequest.getAttribute(
+				JavaConstants.JAVAX_PORTLET_RESPONSE);
+
+		SearchContainer<? extends ContentDashboardItemType> searchContainer =
+			_contentDashboardItemTypeItemSelectorProvider.getSearchContainer(
+				portletRequest, portletResponse, portletURL);
+
+		String portletNamespace = _portal.getPortletNamespace(
+			ContentDashboardPortletKeys.CONTENT_DASHBOARD_ADMIN);
+
+		servletRequest.setAttribute(
+			ContentDashboardItemTypeItemSelectorViewDisplayContext.class.
```

Comment:
I like it, first time I see this way of passing a parameter. Is it a used pattern?

Reply:
Yeahh!!, you can see many examples in Portal.